src/components/LocationSelector/LocationSelector.js

```diff
@@ -0,0 +1,140 @@
+                            sliderLabel={location.radius}
+                            minRadius={minRadius}
+                            maxRadius={maxRadius}
+                            // lots of other props
+                        />
+                    ))}
+                    <Map />
+                </LoadScriptNext>
+            </Modal>
+        </>
+    );
+}
+
+LocationSelector.displayName = 'LocationSelector';
+
+LocationSelector.propTypes = {
+    /** Google api key */
+    apiKey: PropTypes.string.isRequired,
+    /** stores an array of selected location objects */
+    selectedLocations: PropTypes.arrayOf(
+        PropTypes.shape({
+            id: PropTypes.string.isRequired,
+            description: PropTypes.string.isRequired,
+            structuredFormatting: PropTypes.string.isRequired,
+            center: PropTypes.shapeOf({
+                lng: PropTypes.number.isRequired,
+                lan: PropTypes.number.isRequired,
+            }).isRequired,
+            radius: PropTypes.number.isRequired,
+        })
+    ),
+    /** country where search can take place */
+    country: PropTypes.string.isRequired,
+    /** language in which suggestions should be displayed */
+    language: PropTypes.string.isRequired,
+    /** radius measurement unit */
+    radiusUnits: PropTypes.oneOf(['km', 'mi']).isRequired,
+    /** radius label suffix (e.g. km, miles or other language) */
+    radiusUnitDisplayText: PropTypes.string.isRequired,
+    /** min radius value of the slider component */
+    minRadius: PropTypes.number.isRequired,
+    /** max radius value of the slider component */
+    maxRadius: PropTypes.number.isRequired,
+    /** radius step value of the slider component */
+    radiusStep: PropTypes.number.isRequired,
+    /** map center coordinates if nothing has been selected yet */
+    mapCenter: PropTypes.shape({
+        lng: PropTypes.number.isRequired,
+        lat: PropTypes.number.isRequired,
+    }).isRequired,
+    /** map localize */
+    mapRegion: PropTypes.string,
+    /** map default zoom value */
+    mapZoom: PropTypes.number.isRequired,
+    /** placeholder for LocationAutocomplete field */
+    autocompletePlaceholder: PropTypes.string.isRequired,
+    /** placeholder for empty LocationAutocomplete list */
+    noSuggestionsPlaceholder: PropTypes.string.isRequired,
+    /** string to be displayed in FieldWrapper when the modal is closed, but locations are selected */
+    selectionPlaceholder: PropTypes.string.isRequired,
+    /** placeholder for FieldWrapper when there's no locations selected */
+    mainPlaceholder: PropTypes.string.isRequired,
+    /** label for the Done button */
+    doneLabel: PropTypes.string.isRequired,
+    /** show country name in autocomplete suggestions */
+    showCountryInSuggestions: PropTypes.bool,
+    /** label to be used on the clear all button */
+    removeAllLabel: PropTypes.string.isRequired,
```

Comment:
maybe we should call it the same as in `FieldWrapper` to be consistent: `clearLabel`. Or rename it there as well